iib/workers/tasks/build.py

```diff
@@ -720,6 +721,66 @@ def _verify_index_image(
         )
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str db_path: path to the index database
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = 'INSERT INTO properties (type, value, operatorbundle_name, operatorbundle_version, \
+        operatorbundle_path) VALUES (?, ?, ?, ?, ?);'
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property['type'],
+            property['value'],
+            property['operatorbundle_name'],
+            property['operatorbundle_version'],
+            property['operatorbundle_path'],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+def _requires_max_ocp_version(bundle):
+    """
+    Check if the bundle requires the olm.maxOpenShiftVersion property.
+
+    This property is required for bundles using deprecated APIs that don't already
+    have the olm.maxOpenShiftVersion property set.
+
+    :param str bundle: a string representing the bundle pull specification
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '-b',
+        'podman',
+    ]
+    result = run_cmd(cmd, strict=False)
+    if result:
+        output = json.loads(result)
+        # check if the bundle validation failed
+        if not output['passed']:
+            # check if the failure is due to the presence of deprecated APIs
+            # and absence of the 'olm.maxOpenShiftVersion' property
+            # Note: there is no other error in the sdk that mentions this field
+            for msg in output['outputs']:
+                if 'olm.maxOpenShiftVersion' in msg['message']:
```

Comment:
ihmo we need to improve it. 
The validator can return an error with that if we have the property set and its value is > 4.8 in this case. 
So, before inset the data in the database we need to check if that already exists and if yes, then update it.

Reply:
I think what we had discussed was that we would not change what the operator author has explicitly set.

Reply:
Note: this logic is only for new bundles being added not old bundles (the one-time script will take care of those)

Reply:
We are adding it to be called in a method to do the build of the bundle. It might be called after as well. What happens if it is called twice (rebuild for e.g.)?  What happens if the user informed of this value in the CSV? This property is new but we are adding this information in the docs then, operator authors might start to use it and they can add invalid information and the error will be raised as well. 

In this way, IMHO we need to cover these scenarios. So, could we not only add automatically the values when we know that it does not have anything set in the index DB? If the user informed that will not be that in the index DB too? 

- Check the error 
- Before insert do a select in the index DB.
- If we found the property set already we need to skip and track this scenario.